File: app/src/main/java/peter/util/searcher/fragment/BookmarkFragment.java
package peter.util.searcher.fragment;


import androidx.appcompat.widget.SearchView;
import androidx.fragment.app.Fragment;

/**
 * 书签fragment父类
 * <p>
 * Created by peter on 2017/9/28.
 */

public abstract class BookmarkFragment extends Fragment {

    protected SearchView mSearchView;

    public boolean needCloseSearchView() {
        if (mSearchView != null && !mSearchView.isIconified()) {//open
            mSearchView.setIconified(true);
            return true;
        }
        return false;
    }

}

File: app/src/main/java/peter/util/searcher/fragment/BookmarkFragment.kt
package peter.util.searcher.fragment

import android.support.v4.app.Fragment
import android.support.v7.widget.SearchView

/**
 * 书签fragment父类
 *
 *
 * Created by peter on 2017/9/28.
 */

abstract class BookmarkFragment : Fragment() {

    protected var mSearchView: SearchView? = null

    fun needCloseSearchView(): Boolean {
        if (mSearchView != null && !mSearchView!!.isIconified) {//open
            mSearchView!!.isIconified = true
            return true
        }
        return false
    }

}
